src/compiler/checker.ts

```diff
@@ -11846,16 +11885,14 @@ namespace ts {
             }
 
             // TS 1.0 spec (April 2014): 8.3.2
-            // Constructors of classes with no extends clause may not contain super calls, whereas
-            // constructors of derived classes must contain at least one super call somewhere in their function body.
+            // Constructors of classes with no extends clause and constructors of classes that extends null may not contain super calls,
+            // whereas constructors of derived classes must contain at least one super call somewhere in their function body.

```

Comment:
[This isn't currently in the spec](https://github.com/Microsoft/TypeScript/blob/116bce37ea71a5b4422dbba72454f640b944c4b2/doc/spec.md#832-super-calls), so revert this and file a spec bug.